upup/pkg/fi/cloudup/awstasks/subnet.go

```diff
@@ -267,6 +286,41 @@ func (_ *Subnet) RenderAWS(t *awsup.AWSAPITarget, a, e, changes *Subnet) error {
 		}
 	}
 
+	if changes.ResourceBasedNaming != nil {
+		hostnameType := ec2.HostnameTypeIpName
+		if *changes.ResourceBasedNaming {
+			hostnameType = ec2.HostnameTypeResourceName
+		}
+		request := &ec2.ModifySubnetAttributeInput{
+			SubnetId:                       e.ID,
+			PrivateDnsHostnameTypeOnLaunch: &hostnameType,
+		}
+		_, err := t.Cloud.EC2().ModifySubnetAttribute(request)
+		if err != nil {
+			return fmt.Errorf("error modifying hostname type: %w", err)
+		}
+
+		request = &ec2.ModifySubnetAttributeInput{
```

Comment:
I discovered that in ipv6-native subnets, A records cannot be enabled. So here we need to know if the subnet has that enabled or not.

Reply:
kOps doesn't support ipv6-native subnets yet. When we add that support, we will have to update this logic appropriately.